ui/components-react/src/components-react/filter-builder/FilterBuilderState.ts

```diff
@@ -158,6 +158,15 @@ export class PropertyFilterBuilderActions {
       const rule = findRule(state.rootGroup, path);
       if (!rule) return;
       if (isUnaryPropertyFilterOperator(operator)) rule.value = undefined;
+      if (
+        (!isOperatorEqualOrIsNotEqual(rule.operator) ||
+          !isOperatorEqualOrIsNotEqual(operator)) &&
+        (!isInequalityOperator(rule.operator) ||
+          !isInequalityOperator(operator)) &&
+        operator !== rule.operator
+      ) {
```

Comment:
This condition is quite hard to understand with all these `!` operators. Maybe you could add one more function that would check if operators are similar (both operators are `equal/notEqual` or both operators are one of the `>`, `>=`, `<`, `<=`). That way you could simplify this condition:
```suggestion
      if (operator !== rule.operator && !areOperatorsSimilar(operator, rule.operator)) {
```

Reply:
fixed